File: src/main/java/de/moritzf/quota/idea/QuotaSettingsState.java
package de.moritzf.quota.idea;

import com.intellij.openapi.application.ApplicationManager;
import com.intellij.openapi.components.PersistentStateComponent;
import com.intellij.openapi.components.Service;
import com.intellij.openapi.components.State;
import com.intellij.openapi.components.Storage;
import org.jetbrains.annotations.NotNull;

/**
 * Persistent plugin settings shared at application scope.
 */
@State(name = "OpenAiUsageQuotaSettings", storages = @Storage("openai-usage-quota.xml"))
@Service(Service.Level.APP)
public final class QuotaSettingsState implements PersistentStateComponent<QuotaSettingsState> {
    public int refreshMinutes = 5;
    public String statusBarDisplayMode = QuotaDisplayMode.ICON_ONLY.name();

    public static QuotaSettingsState getInstance() {
        return ApplicationManager.getApplication().getService(QuotaSettingsState.class);
    }

    @Override
    public @NotNull QuotaSettingsState getState() {
        return this;
    }

    @Override
    public void loadState(@NotNull QuotaSettingsState state) {
        this.refreshMinutes = state.refreshMinutes;
        this.statusBarDisplayMode = QuotaDisplayMode.fromStorageValue(state.statusBarDisplayMode).name();
    }

    public @NotNull QuotaDisplayMode getStatusBarDisplayMode() {
        return QuotaDisplayMode.fromStorageValue(statusBarDisplayMode);
    }

    public void setStatusBarDisplayMode(@NotNull QuotaDisplayMode displayMode) {
        this.statusBarDisplayMode = displayMode.name();
    }
}

File: src/main/kotlin/de/moritzf/quota/idea/QuotaSettingsState.kt
package de.moritzf.quota.idea

import com.intellij.openapi.application.ApplicationManager
import com.intellij.openapi.components.PersistentStateComponent
import com.intellij.openapi.components.Service
import com.intellij.openapi.components.State
import com.intellij.openapi.components.Storage

/**
 * Persistent plugin settings shared at application scope.
 */
@State(name = "OpenAiUsageQuotaSettings", storages = [Storage("openai-usage-quota.xml")])
@Service(Service.Level.APP)
class QuotaSettingsState : PersistentStateComponent<QuotaSettingsState> {
    var refreshMinutes: Int = 5
    var statusBarDisplayMode: String = QuotaDisplayMode.ICON_ONLY.name

    override fun getState(): QuotaSettingsState = this

    override fun loadState(state: QuotaSettingsState) {
        refreshMinutes = state.refreshMinutes
        statusBarDisplayMode = QuotaDisplayMode.fromStorageValue(state.statusBarDisplayMode).name
    }

    fun displayMode(): QuotaDisplayMode = QuotaDisplayMode.fromStorageValue(statusBarDisplayMode)

    fun setDisplayMode(displayMode: QuotaDisplayMode) {
        statusBarDisplayMode = displayMode.name
    }

    companion object {
        @JvmStatic
        fun getInstance(): QuotaSettingsState {
            return ApplicationManager.getApplication().getService(QuotaSettingsState::class.java)
        }
    }
}
